stdlib/option.mc

```diff
@@ -94,6 +94,21 @@ let optionMapM: (a -> Option b) -> [a] -> Option [b] = lam f. lam l.
 utest optionMapM (lam x. if gti x 2 then Some x else None ()) [3, 4, 5] with Some [3, 4, 5]
 utest optionMapM (lam x. if gti x 2 then Some x else None ()) [2, 3, 4] with None ()
 
+let optionFoldlM: (a -> b -> Option a) -> a -> [b] -> Option a = lam f.
+  recursive let recur = lam a. lam bs.
+    match bs with [b] ++ bs then
+      match f a b with Some a then
+        recur a bs
+      else None ()
+    else Some a
```

Comment:
Should probably be
```
else match bs with [] then
  Some a
else never
```
?

Reply:
This probably depends on what policy we want to have for exhaustive pattern matching. Writing the extra match and ending with a `never` would trigger our exhaustiveness analysis, but the semantics are otherwise the same. This is essentially equivalent with `match bs with | [b] ++ bs -> ... | _ -> ...`, while using `never` would instead be equivalent with `match bs with | [b] ++ bs -> ... | [] -> ...`. I would probably normally argue for the latter, I've just been lazy here.

Maybe worth deciding on a convention to follow generally?

Reply:
Probably a good idea to discuss at the next meeting.